pkgs/development/libraries/mpich/default.nix

```diff
@@ -1,24 +1,50 @@
-{ lib, stdenv, fetchurl, perl, gfortran
-,  openssh, hwloc
+{ stdenv, lib, fetchurl, perl, gfortran
+,  openssh, hwloc, autoconf, automake, libtool
+# device options are ch3 or ch4
```

Comment:
Please add an assert to check if it is either ch3 or ch4.

Reply:
Isn't the assert here a bit redundant? If you pick the wrong option configure will simply fail. With an assert I need to keep that in line with all upstream options increasing the maintenance burden.

Reply:
Assert is quite useful in this scenario because it fails quite early on but fine with me if you think failing in configure is early enough.

Reply:
I agree that an assert can be useful. For example for the backend option it is, since it would otherwise fail quietly (or not fail properly at all). Here I think we can we leave the check to configure, which will fail quite early.